### PR TITLE
add apptheme field to use data uri instead of blob object

### PR DESCRIPTION
**Pull request by jwunderl**
re: https://github.com/microsoft/pxt-microbit/issues/3674, will toggle this on with https://github.com/microsoft/pxt-microbit/blob/master/pxtarget.json#L563 (and if needed for iosapp)

### CODE REVIEW

File: localtypings/pxtarget.d.ts [hunk 1]
```diff
@@ -405,6 +405,7 @@ declare namespace pxt {
         identity?: boolean; // login with identity providers
         assetEditor?: boolean; // enable asset editor view (in blocks/text toggle)
         disableMemoryWorkspaceWarning?: boolean; // do not warn the user when switching to in memory workspace
+        disableBlobObjectDownload?: boolean; // use data uri downloads instead of object urls
```
Comment (pelikhan) [inline localtypings/pxtarget.d.ts:408]: why is this a flag for an entire editor, isn't this browser/host specific?

Comment (jwunderl) [reply localtypings/pxtarget.d.ts:408]: It is host specific; I was going to use the query variants to toggle it, but if preferred can just change this to check for the query parameters in the browserdownload function?

Comment (pelikhan) [reply localtypings/pxtarget.d.ts:408]: would be best to detect this at runtime one way or the other i guess. This is for electron/ios?

Comment (jwunderl) [reply localtypings/pxtarget.d.ts:408]: for the android / probably ios mobile apps; for android users going through the browser we still would want the blob object download, it's just not as accessible to the app to intercept / store that way

**Review by abchatra — APPROVED** [review]

**Pull request merged**